### PR TITLE
Support Databricks 3.0.1 with new build profiles

### PR DESCRIPTION
**Pull request by tgravescs**
relates to https://github.com/NVIDIA/spark-rapids/issues/3224

Add support for databricks 3.0.1 with new build profiles.  In this case so prevent having to copy all the dependencies for data bricks 3.1.1 into a profile for 3.0.1 I made another profiles dbdeps that is triggered by property -Ddatabricks.
If we want to remove this we can and we can just copy them.

I also ran into a bug with the scala-maven-plugin where I had to go back to an older version.  

Also clean up some build warnings.

build command is like:

`mvn -Ddatabricks -Dbuildver=301db package
`

### CODE REVIEW

**Comment by tgravescs** [comment]
build

**Review by jlowe — COMMENTED** [review]
For some reason the use of `S{spark.version}` for all the non-Spark jars struck me this time, surprised I didn't notice it before.  This is done in a lot of places, so I didn't put a comment for  each one.  Not sure it absolutely has to have a comment, but it sure did look weird and wrong when I noticed it.

File: shuffle-plugin/pom.xml [hunk 1]
```diff
@@ -128,24 +128,23 @@
             </build>
         </profile>
        <profile>
-            <id>release311db</id>
+            <id>dbdeps</id>
             <activation>
                 <property>
-                    <name>buildver</name>
-                    <value>311db</value>
+                    <name>databricks</name>
```
[the remark this answers is not in the document]
Comment (tgravescs) [reply shuffle-plugin/pom.xml:134]: I didn't update the build script yet to build using the new way.  the build script simply can install the dependencies and you manually run mvn after that. I was thinking we would do that once we switch over and want things to run automated.    I can certainly do it here as well if you prefer?

[the remark this answers is not in the document]
Comment (jlowe) [reply shuffle-plugin/pom.xml:134]: No, that's fine.  We can leave it unspecified until we switch.

**Review by jlowe — APPROVED** [review]

**Comment by tgravescs** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**